### PR TITLE
Logging: set Trace in trace instead of label

### PR DESCRIPTION
**Pull request by andreamlin**
Fixes https://github.com/GoogleCloudPlatform/google-cloud-java/issues/3280.

### CODE REVIEW

**Comment by andreamlin** [comment]
@pongad @joaoandremartins 
PTAL

File: google-cloud-clients/google-cloud-logging/src/main/java/com/google/cloud/logging/TraceLoggingEnhancer.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public TraceLoggingEnhancer(String prefix) {
   /**
    * Set the Trace ID associated with any logging done by the current thread.
    *
-   * @param id The traceID
+   * @param id The traceID, in the form projects/[PROJECT_ID]/traces/[TRACE_ID]
```
Comment (joaoandremartins) [inline google-cloud-clients/google-cloud-logging/src/main/java/com/google/cloud/logging/TraceLoggingEnhancer.java]: Have you considered also getting the project ID as an argument and building the `projects/[PROJECT_ID]/traces/[TRACE_ID]` format from there?

Comment (andreamlin) [reply google-cloud-clients/google-cloud-logging/src/main/java/com/google/cloud/logging/TraceLoggingEnhancer.java]: The convention for that in this repo would be to create a `TraceName` class that implements `ResourceName` and has a Builder pattern (or static constructor) and a formatting `toString()` method.

Comment (joaoandremartins) [reply google-cloud-clients/google-cloud-logging/src/main/java/com/google/cloud/logging/TraceLoggingEnhancer.java]: OK, LGTM then

**Pull request merged**